lib/Common/Memory/HeapBucket.inl

```diff
@@ -7,25 +7,17 @@
 template <typename TBlockType>
 template <ObjectInfoBits attributes, bool nothrow>
 __inline char *
-HeapBucketT<TBlockType>::RealAlloc(Recycler * recycler, size_t sizeCat)
+HeapBucketT<TBlockType>::RealAlloc(Recycler * recycler, size_t sizeCat, size_t size)
 {
     Assert(sizeCat == this->sizeCat);
 
     char * memBlock = allocatorHead.InlinedAlloc<(ObjectInfoBits)(attributes & InternalObjectInfoBitMask)>(recycler, sizeCat);
 
     if (memBlock == nullptr)
     {
-        memBlock = SnailAlloc(recycler, &allocatorHead, sizeCat, attributes, nothrow);
+        memBlock = SnailAlloc(recycler, &allocatorHead, sizeCat, size, attributes, nothrow);
         Assert(memBlock != nullptr || nothrow);
     }
-    else
-    {
-#ifdef RECYCLER_PAGE_HEAP
-        Assert(allocatorHead.heapBlock == nullptr || !allocatorHead.heapBlock->InPageHeapMode());
-#else
-        Assert(allocatorHead.heapBlock == nullptr);
```

Comment:
let me add non-pageheap part back

---
In reply to: [60987129](https://github.com/Microsoft/ChakraCore/pull/711#discussion_r60987129) [](ancestors = 60987129)